allennlp/models/encoder_decoders/nlvr/nlvr_semantic_parser.py

```diff
@@ -362,137 +352,30 @@ def _embed_actions(self, actions: List[List[ProductionRuleArray]]) -> Tuple[torc
             Has shape ``(num_unique_actions, action_embedding_dim)``.
         action_map : ``Dict[Tuple[int, int], int]``
             Maps ``(batch_index, action_index)`` in the input action list to ``action_index`` in
-            the ``action_embeddings`` tensor.
-        initial_action_embedding : ``torch.Tensor``
-            Has shape ``(action_embedding_dim,)``.  An embedding for the initial action input.
-            This needs to be computed here, because we don't keep around the nonterminal
-            embeddings.  We do this by creating a fake rule "0 -> START", where the LHS embedding
-            is a vector of zeros, and the RHS embedding is the START symbol embedding.
+            the ``action_embeddings`` tensor.  All non-embeddable actions get mapped to `-1` here.
```

Comment:
That's just padding, right?

Reply:
Not for NLVR.  It shouldn't ever happen at all - padded actions are skipped.  I changed this.